data/pools/transactionPool.go

```diff
@@ -352,81 +251,55 @@ func (pool *TransactionPool) OnNewBlock(block bookkeeping.Block) {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
 
-	var stats telemetryspec.ProcessBlockMetrics
-
-	remove := make(map[transactions.Txid]error, 0)
-
-	// collect transactions that appear in the last block
-	payset, err := block.DecodePayset()
-	if err != nil {
-		return
-	}
-
-	for _, tx := range payset {
-		txid := tx.ID()
-		_, ok := pool.pendingTxns[txid]
-		if ok {
-			remove[txid] = nil
-			stats.KnownCommittedCount++
+	// Adjust the pool fee threshold.  The rules are:
+	// - If there was less than one full block in the pool, reduce
+	//   the multiplier by 2x.  It will eventually go to 0, so that
+	//   only the flat MinTxnFee matters if the pool is idle.
+	// - If there were less than two full blocks in the pool, keep
+	//   the multiplier as-is.
+	// - If there were two or more full blocks in the pool, grow
+	//   the multiplier by 2x (or increment by 1, if 0).
+	switch pool.numPendingWholeBlocks {
```

Comment:
Shouldn't this happen after `recomputeBlockEvaluator`?

Reply:
Also, should this be guarded by the conditional `if pool.pendingBlockEvaluator == nil || block.Round() >= pool.pendingBlockEvaluator.Round() {`?

Reply:
We want to measure the size of the pool before the next block's worth of transactions get subtracted away.  Moving this code to after `recomputeBlockEvaluator` would mean we have no way to distinguish the case when the pool is under-utilized (`numPendingWholeBlocks == 0`, where we want to reduce the threshold) from the case when the pool is about in steady-state (`numPendingWholeBlocks == 1`, where we want to keep the threshold as-is).

Reply:
Moving it into the guarded block sounds reasonable if it seems clearer.  I'll do that.

The reason I don't think it matters is that it would only affect the computation of `feeThresholdMultiplier` during catchup (when that guarded block might not fire), and the threshold is sort-of meaningless during catchup anyway.